CMakeLists.txt

```diff
@@ -347,6 +347,19 @@ endif ()
 # ADDING SOURCE & TOOLS
 #==============================================================================
 
+# https://github.com/E3SM-Project/scorpio/pull/439 moved MPI detection to subprojects.
+# While this is a good practice, a confirmed CMake regression (3.22.0 or higher) causes this approach to fail on Cray systems.
+# References:
+# - https://github.com/E3SM-Project/scorpio/issues/517
+# - https://discourse.cmake.org/t/regression-cmake-3-22-fails-to-find-mpi-on-cray-systems-reproducible-on-frontier-supercomputer/13045
+#
+# Until CMake developers fix this regression, a confirmed workaround is to add a redundant find_package(MPI) call at the root level.
```

Comment:
I think we should keep the history of the issue in the github PR/issue. The comment here should just explain that its a hack for Cray systems + cmake versions etc (and why we need this hack). We should also move the workarounds etc to the PR comments (or issue description).

Reply:
OK, I have updated the comment here.